test-result-summary-client/src/PerfCompare/lib/JenkinsRunJSON.js

```diff
@@ -26,25 +23,24 @@ export default class JenkinsRunJSON {
                 // new variant
                 if (parsedVariantsCommon[curVariantObjectId] === undefined) {
                     parsedVariantsCommon[curVariantObjectId] = {
-                        product: (this.runJSON.testInfo[0].tests[k].benchmarkProduct === undefined) ? null : this.runJSON.testInfo[0].tests[k].benchmarkProduct,
-                        result: (this.runJSON.testInfo[0].tests[k].testResult === undefined) ? null : this.runJSON.testInfo[0].tests[k].testResult,
-                        benchmark: (this.runJSON.testInfo[0].tests[k].benchmarkName === undefined) ? null : this.runJSON.testInfo[0].tests[k].benchmarkName,
-                        variant: (this.runJSON.testInfo[0].tests[k].benchmarkVariant === undefined) ? null : this.runJSON.testInfo[0].tests[k].benchmarkVariant,
-                        machine: (this.runJSON.testInfo[0].machine === undefined) ? null : this.runJSON.testInfo[0].machine,
-                        metrics: (this.runJSON.testInfo[0].tests[k].testData.metrics === undefined) ? [] : this.runJSON.testInfo[0].tests[k].testData.metrics
+                        product: (this.runJSON.testInfo[0].aggregateInfo[k].benchmarkProduct === undefined) ? null : this.runJSON.testInfo[0].aggregateInfo[k].benchmarkProduct,
+                        benchmark: (this.runJSON.testInfo[0].aggregateInfo[k].benchmarkName === undefined) ? null : this.runJSON.testInfo[0].aggregateInfo[k].benchmarkName,
+                        variant: (this.runJSON.testInfo[0].aggregateInfo[k].benchmarkVariant === undefined) ? null : this.runJSON.testInfo[0].aggregateInfo[k].benchmarkVariant,
+                        metrics: (this.runJSON.testInfo[0].aggregateInfo[k].metrics === undefined) ? [] : this.runJSON.testInfo[0].aggregateInfo[k].metrics,
+                        testsData: (this.runJSON.testInfo[0].tests === undefined) ? undefined : this.runJSON.testInfo[0].tests[0].testData.metrics
                     };
 
                 // variant already exists
                 } else {
                     // loop over the current metrics
-                    for (let x = 0; x < this.runJSON.testInfo[0].tests[k].testData.metrics.length; x++) {
+                    for (let x = 0; x < this.runJSON.testInfo[0].aggregateInfo[k].metrics.length; x++) {
 
                         // loop over existing metrics
                         for (let y = 0; y < parsedVariantsCommon[curVariantObjectId].metrics.length; y++) {
 
                             // matching metric found
-                            if (this.runJSON.testInfo[0].tests[k].testData.metrics[x].name === parsedVariantsCommon[curVariantObjectId].metrics[y].name) {
-
+                            if (this.runJSON.testInfo[0].aggregateInfo[k].metrics[x].name === parsedVariantsCommon[curVariantObjectId].metrics[y].name) {
+                                  this.parsedVariants.push(parsedVariantsCommon[curVariantObjectId].metrics[y].value)
```

Comment:
Not sure whether we need this! Could you please add some comment in the code for the purpose of storing that data? We are already storing the relevant info at the end of this function: `this.parsedVariants.push(curVariantObject);`.